lib/Controller/ThreadController.php

```diff
@@ -111,4 +121,40 @@ public function delete(int $id): JSONResponse {
 
 		return new JSONResponse();
 	}
+
+	/**
+	 * @NoAdminRequired
+	 *
+	 * @param int $id
+	 *
+	 * @return JSONResponse
+	 */
+	public function summarize(int $id): JSONResponse {
+		try {
+			$message = $this->mailManager->getMessage($this->currentUserId, $id);
+			$mailbox = $this->mailManager->getMailbox($this->currentUserId, $message->getMailboxId());
+			$account = $this->accountService->find($this->currentUserId, $mailbox->getAccountId());
+		} catch (DoesNotExistException $e) {
+			return new JSONResponse([], Http::STATUS_FORBIDDEN);
+		}
+		if (empty($message->getThreadRootId())) {
+			return new JSONResponse([], Http::STATUS_NOT_FOUND);
+		}
+		$thread = $this->mailManager->getThread($account, $message->getThreadRootId());
+		try {
+			$summary = $this->aiIntergrationsService->summarizeThread(
+				$message->getThreadRootId(),
+				$thread,
+				$this->currentUserId,
+			);
+		} catch (\Throwable $e) {
+			$this->logger->error('Summarizing thread failed: ' . $e->getMessage(), [
+				'exception' => $e,
+			]);
+			return new JSONResponse([], Http::STATUS_NO_CONTENT);
```

Comment:
Hide the whole summary box in case of error. Currently, it will be rendered with no text which looks weird.

Reply:
Solid error handling will be important when someone uses the llm app as text processing backend and it runs into a timeout :+1: